### PR TITLE
WT-8714 Support linking CMake test binaries with a relative RPATH

### PR DESCRIPTION
**Pull request by Siddhartha8899**

### CODE REVIEW

File: test/ctest_helpers.cmake [hunk 1]
```diff
@@ -77,6 +77,24 @@ function(create_test_executable target)
         target_link_libraries(${target} ${CREATE_TEST_LIBS})
     endif()
 
+    if(WT_LINUX OR WT_DARWIN)
```
Comment (etienneptl) [inline test/ctest_helpers.cmake:80]: Should do it for Windows too?

Comment (Siddhartha8899) [reply test/ctest_helpers.cmake:80]: Good point, after doing some research I found out that Windows does things completely differently and does not have a concept of rpath. Windows shared libraries are always looked up in directories that are in the current PATH, or by adding libraries to the PATH before running the program. 

That said, if need be we can extend this to Windows as well. I will not worry about it for now as it requires more investigation and is out of the scope of this ticket.

File: test/ctest_helpers.cmake [hunk 2]
```diff
@@ -77,6 +77,24 @@ function(create_test_executable target)
...
+        set(origin_linker_variable)
+        if(WT_LINUX)
+            set(origin_linker_variable "\\$ORIGIN")
+        elseif(WT_DARWIN)
+            set(origin_linker_variable "@loader_path")
+        endif()
+        file(RELATIVE_PATH relative_rpath ${test_binary_dir} ${CMAKE_BINARY_DIR})
+        set_target_properties(${target}
+            PROPERTIES
+                BUILD_WITH_INSTALL_RPATH FALSE
```
Comment (etienneptl) [inline test/ctest_helpers.cmake:93]: Please add a comment on why this value is set to `FALSE` as it might be useful for anyone who reads this code.

Comment (Siddhartha8899) [reply test/ctest_helpers.cmake:94]: Added in [e02da67](https://github.com/wiredtiger/wiredtiger/pull/7524/commits/e02da67e0aebe0cf8031d83aaaa0fa69a9be46e5)

**Review by korteland — APPROVED** [review]
Looks good to me!

**Pull request merged**